### PR TITLE
Builtin coverage

### PR DESCRIPTION
**Pull request by ahasikos**
# Description

Increases test coverage of the builtin package and specifically:

- authority.go
- gen/bindata.go

Fixes #[190](https://github.com/vechain/protocol-board-repo/issues/109) 

## Type of change

Adds code coverage tests

# How Has This Been Tested?

`make test`

**Test Configuration**:

* Go Version:  go1.22.1 darwin/arm64
* Hardware: MBP M3
* Docker Version: -

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] New and existing E2E tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have not added any vulnerable dependencies to my code

### CODE REVIEW

File: builtin/authority/authority_test.go [hunk 1]
```diff
@@ -62,6 +62,15 @@ func TestAuthority(t *testing.T) {
 		{M(aut.Candidates(&big.Int{}, thor.InitialMaxBlockProposers)), M(
 			[]*Candidate{{p2, p2, thor.Bytes32{}, true}, {p3, p3, thor.Bytes32{}, true}}, nil,
 		)},
+		{M(aut.AllCandidates()), M([]*Candidate{
+			//{p1, p1, thor.Bytes32{}, false},
```
Comment (paologalligit) [inline builtin/authority/authority_test.go:66]: Is this deletable or should be uncommented?

Comment (ahasikos) [reply builtin/authority/authority_test.go]: Line deleted.

**Review by paologalligit — APPROVED** [review]
LGTM, just a comment about a doubt I have. I saw `bindata.go` is an auto-generated [file](https://github.com/vechain/thor/blob/master/builtin/gen/bindata.go#L1). Maybe we could have excluded this kind of file to be counted in the coverage percentage. It feels like we're testing the tool that has generated the file, which I suppose is an external tool.

**Comment by ahasikos** [comment]
@paologalligit I agree. We need to decide on how we exclude either functions or files for being tested.

**Comment by libotony** [comment]
> @paologalligit I agree. We need to decide on how we exclude either functions or files from being tested.

Could you remove the generated file's test and let's find a way to improve the codecov strategy?

**Review by libotony — APPROVED** [review]

**Pull request merged**